source/common/quic/envoy_quic_server_session.cc

```diff
@@ -109,29 +108,23 @@ void EnvoyQuicServerSession::SetDefaultEncryptionLevel(quic::EncryptionLevel lev
   if (level != quic::ENCRYPTION_FORWARD_SECURE) {
     return;
   }
-  maybeCreateNetworkFilters();
   // This is only reached once, when handshake is done.
   raiseConnectionEvent(Network::ConnectionEvent::Connected);
 }
 
 bool EnvoyQuicServerSession::hasDataToWrite() { return HasDataToWrite(); }
 
-void EnvoyQuicServerSession::OnTlsHandshakeComplete() {
-  quic::QuicServerSessionBase::OnTlsHandshakeComplete();
-  maybeCreateNetworkFilters();
-  raiseConnectionEvent(Network::ConnectionEvent::Connected);
+const quic::QuicConnection* EnvoyQuicServerSession::quicConnection() const {
+  return initialized_ ? connection() : nullptr;
```

Comment:
Is it valid to get here when initialized_ is false? Should we try harder to ensure that Initialize() is called?